r/R/dplyr-funcs.R

```diff
@@ -58,15 +58,34 @@ NULL
 #' @keywords internal
 #'
 register_binding <- function(fun_name, fun, registry = nse_funcs) {
-  name <- gsub("^.*?::", "", fun_name)
-  namespace <- gsub("::.*$", "", fun_name)
+  qualified_name <- fun_name
+  if (qualified_name == "::") {
+    unqualified_name <- "::"
+  } else {
+    unqualified_name <- gsub("^.*?::", "", qualified_name)
+  }
 
-  previous_fun <- if (name %in% names(registry)) registry[[name]] else NULL
+  previous_fun <- if (unqualified_name %in% names(registry)) registry[[unqualified_name]] else NULL
+
+  # if th unqualified name exists in the register, warn
+  if (!is.null(fun) && !is.null(previous_fun)) {
+    warn(
+      paste0(
+        "A \"",
+        unqualified_name,
+        "\" binding already exists in the register and will be overwritten.")
+    )
+  }
 
+  # if fun is NULL remove entries from the function registry
```

Comment:
Is this a real use case? Do we test this?

Reply:
Not 💯  convinced it is a real use case. We test it [here](https://github.com/dragosmg/arrow/blob/eb5d35a735bb97e706e6c4a0f0e7f6f06bc3cb7f/r/tests/testthat/test-dplyr-funcs.R#L27-L28).

Reply:
I plan on working on [ARROW-14071: [R] Try to arrow_eval user-defined functions](https://issues.apache.org/jira/browse/ARROW-14071) next, so the ability to remove a binding could be useful there, but it might not be. Since it doesn't play a role in this PR, I think we should remove it and add it back in if needed. I will remove it and mark this as resolved. Let me know if you think otherwise.